### PR TITLE
BUG: concat bool+int with ArrayManager

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/internals/concat.py [hunk 1]
```diff
@@ -123,12 +123,16 @@ def concat_arrays(to_concat: list) -> ArrayLike:
...
-    if not single_dtype:
-        target_dtype = find_common_type([arr.dtype for arr in to_concat_no_proxy])
-    else:
+    if single_dtype:
         target_dtype = to_concat_no_proxy[0].dtype
+    elif all(x.kind in ["i", "u", "b"] and isinstance(x, np.dtype) for x in dtypes):
+        # GH#42092
+        target_dtype = np.find_common_type(list(dtypes), [])
+    else:
+        target_dtype = find_common_type([arr.dtype for arr in to_concat_no_proxy])
```
Comment (jreback) [inline pandas/core/internals/concat.py:132]: maybe should just modify fond_common_dtyoe itself to handle both cases

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/concat.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/concat.py:132]: find_common_type has the desired long-term behavior; we need to deprecate the behavior here to match in the future

Comment (jreback) [reply pandas/core/internals/concat.py:132]: i don't understand your comment

Comment (jbrockmendel) [reply pandas/core/internals/concat.py:132]: find_common_type is used elsewhere, and we don't want to change the behavior of the other usages.  so find_common_type shouldn't be changed.

This is fixing a regression #42092.  Eventually we want to change the behavior back to what it is in master, but doing that right will require a deprecation cycle.

**Comment by jbrockmendel** [comment]
(this is whats needed to get the CI back to green)

File: pandas/core/internals/concat.py (hunk 1 above)
Comment (jreback) [reply pandas/core/internals/concat.py:132]: kk got it.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**